pythran/pythonic/numpy/ndarray/reshape.hpp

```diff
@@ -27,7 +28,18 @@ namespace numpy
                             types::ndarray<T, NpS>>::type
     reshape(types::ndarray<T, pS> const &expr, NpS const &new_shape)
     {
-      long where = sutils::find(new_shape, -1);
+      std::function<bool(long, long)> comp = [](long a, long b) {
+        return a <= b;
+      };
+      long where = sutils::find(
+          new_shape, -1,
+          std::integral_constant<size_t, std::tuple_size<NpS>::value>(), comp);
```

Comment:
why do you start from the end?

Reply:
That's what you did... I didn't change that part